src/auto-evo/mutation_strategy/AddOrganelleAnywhere.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace AutoEvo;
+
+using System;
+using System.Collections.Frozen;
+using System.Collections.Generic;
+using System.Linq;
+
+public class AddOrganelleAnywhere : IMutationStrategy<MicrobeSpecies>
+{
+    private readonly CommonMutationFunctions.Direction direction;
+    private readonly FrozenSet<OrganelleDefinition> allOrganelles;
+
+    public AddOrganelleAnywhere(Func<OrganelleDefinition, bool> criteria, CommonMutationFunctions.Direction direction
+        = CommonMutationFunctions.Direction.Neutral)
+    {
+        allOrganelles = SimulationParameters.Instance.GetAllOrganelles().Where(criteria).ToFrozenSet();
+        this.direction = direction;
+    }
+
+    public bool Repeatable => true;
+
+    public static AddOrganelleAnywhere ThatUseCompound(Compound compound, CommonMutationFunctions.Direction direction
+        = CommonMutationFunctions.Direction.Neutral)
+    {
+        return new AddOrganelleAnywhere(organelle => organelle.RunnableProcesses
+            .Where(proc => proc.Process.Inputs.ContainsKey(compound)).Any(), direction);
```

Comment:
I think these can be written in a shorter form like: `RunnableProcesses.Any(proc => proc.Process.Inputs.ContainsKey(compound))`.

I won't suggest this yet but if it turns out that constructing the mutation strategies causes many allocations, pre-caching on game startup these differently configured mutation strategies might be the optimal way to do these.

Reply:
If there was a simple way to cache all of the mutation stratiges that are used by GenerateMiche in between Auto-Evo runs I agree that would be ideal. Once created they are readonly so sharing them between threads or even generations should pose no issues.

Reply:
Yeah, maybe they could even be stored in the auto-evo settings object to have all auto-evo runs forever be able to share them.

Reply:
Thinking about it more, what I said above also applies to all of the Selection Pressures except predation (which obviously depends on the spieces in the patch), so it would probably better to store those.